### PR TITLE
DOCSP-22998 create limitations page

### PR DESCRIPTION
**Pull request by davemungo**
NOTE: In the interest of time, sending this to internal and external review together.

[STAGING](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-22998-create-limitations-page/reference/limitations/)

[JIRA](https://jira.mongodb.org/browse/DOCSP-22998)

### CODE REVIEW

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:21]: what does this mean? this bullet is a bit vague

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:25]: Filtered synchronization (taking a subset of data)

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:45]: M shard cluster to N shard cluster

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:57]: typo

what are CEA failures?

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:58]: I don't understand this bullet. Perhaps delete? 

If a customer does not have any unique indexes, it's a standard case of sync/reverse.

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:65]: -Syncing multiple clusters to one destination cluster is not supported
-Syncing one cluster to many destination clusters is not supported

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:66]: This is mentioned above in the sharding section?

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:22]: spacing

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:24]: Move this point about DDL higher up. After "must be on same release"?

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:12]: +1 limitation. The destination cluster must be an empty cluster

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:60]: This one sounds a bit too technical / implementation detail

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:17]: let's remove this for now. There are special cases where they can (e.g. moving to paid version via Atlas or EA), but "please contact your sales rep".

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [reply source/reference/limitations.txt:22]: link write blocking to the /start endpoint

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [inline source/reference/limitations.txt:6]: Can we add a small note below here saying: "Some of the limitations may be removed from this list in the Generally Available version"

**Review by alan-zheng-mdb — CHANGES_REQUESTED** [review]

File: source/reference/limitations.txt
Comment (ldennis) [reply source/reference/limitations.txt:21]: Do we mean primary shard? Should we somehow link https://www.mongodb.com/docs/manual/reference/command/movePrimary/ ?

File: source/reference/limitations.txt
Comment (ldennis) [inline source/reference/limitations.txt:23]:
```suggestion
  source cluster before starting the commit process.
```

File: source/reference/limitations.txt
Comment (ldennis) [reply source/reference/limitations.txt:57]: CEA: change event application
This is a result of the design decision we made to "Only replicate indexes exist on all shards". `mongosync` doesn't replicate [inconsistent Indexes across shards](https://www.mongodb.com/docs/v5.0/tutorial/manage-indexes/#find-inconsistent-indexes-across-shards). As a result, `mongosync` won't try to relax the constraints for inconsistent unique indexes across shard, and so the existence of an uniqueness index in some but not all shards may cause change event application to fail. I think maybe we could somehow link these two limitations together.

File: source/reference/limitations.txt
Comment (ldennis) [inline source/reference/limitations.txt:37]: Should we try our best to link some of these to the server doc?

File: source/reference/limitations.txt
Comment (ldennis) [reply source/reference/limitations.txt:6]: +1

File: source/reference/limitations.txt
Comment (tfogo) [reply source/reference/limitations.txt:57]: We should not say "CEA" in the docs, we don't explain this term anywhere.

File: source/reference/limitations.txt
Comment (jmd-mongo) [inline source/reference/limitations.txt:50]: [question] 

Should this be "The shard key cannot be redfined using ``RefineShardKey`` during resharding." ? Seems like there's  a word missing.

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt:21]: Can we indicate which limitations will be for GA and which for preview where we know. I'd love to indicate that at GA DDL will be supported

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Removed.

The restriction on M to N sync does cover this idea. This particular version seemed to represent a specific migration use case so I thought it best to highlight it.

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt]: Can we say 'applyOps' command and link to https://www.mongodb.com/docs/manual/reference/command/applyOps/? Some people think of "applyOps" as the name of the process where secondary nodes apply operations from the primary during normal steady state replication. I want to make sure we clarify this for users

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt:46]: collation and capped will be supported at GA, but not time series

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt:63]: s/deleted/removed/

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt:65]: I would actually make this two lines:
1. The shard key cannot be refined with RefineShardKey (link to refine shard key docs)
2. The shard key cannot be modified with reshardCollection (link to reshardCollection docs)

File: source/reference/limitations.txt
Comment (judahschvimer) [reply source/reference/limitations.txt]: We have the prerequisite around ensuring unique indexes are re-initial synced if they were created post 4.0. Alan, can you please help how to word this one?

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt:4]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt:23]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt:29]: I don't have that info, maybe @alan-zheng-mdb or @ldennis knows.

We probably shouldn't promise anything that specific anyway to avoid customer reliance issues in case something slips.

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt:55]: As above, we will update this doc for GA to remove limits that are no longer valid

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

Comment (davemungo) [reply source/reference/limitations.txt]: There are no docs for refineShardKey

Done

**Review by davemungo — COMMENTED** [review]
Thanks everyone. This is ready for another pass

**Comment by davemungo** [comment]
@alan-zheng-mdb , @tfogo, @ldennis , @jmd-mongo, @judahschvimer 

Is it ok if @alan-zheng-mdb signs off on this to merge or should I wait for LGTMs from the whole group?

File: source/reference/limitations.txt
Comment (judahschvimer) [reply source/reference/limitations.txt:55]: hmm, I worry that this will give the impression to some people that their use case isn't supported and they won't look back in a month to see that their use case is now supported.

File: source/reference/limitations.txt
Comment (judahschvimer) [inline source/reference/limitations.txt:25]: I think we should link to the FCV docs like https://www.mongodb.com/docs/v6.0/reference/command/setFeatureCompatibilityVersion/ here. FCV isn't so well known of a concept.

File: source/reference/limitations.txt
Comment (judahschvimer) [reply source/reference/limitations.txt:29]: I think there are some features we must support by GA, so I think I'm comfortable saying they'll come for GA. @ldennis and @alan-zheng-mdb , what do you think?

**Review by judahschvimer — COMMENTED** [review]
If @alan-zheng-mdb is good with it, I'm fine, no need to wait on my approval.

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [reply source/reference/limitations.txt:13]: can we make this paragraph with the red call-out box?

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [reply source/reference/limitations.txt:74]: +1

File: source/reference/limitations.txt
Comment (ldennis) [reply source/reference/limitations.txt:71]: Just that the destination won't inherit the zones. We only replicate data.

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [reply source/reference/limitations.txt:25]: fCV or FCV? I've seen it a few different ways across the docs. 

Ok deleting the acronym as well

File: source/reference/limitations.txt
Comment (alan-zheng-mdb) [reply source/reference/limitations.txt:55]: see comment above

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Updated with "currently ..." as suggested above

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

File: source/reference/limitations.txt
Comment (davemungo) [reply source/reference/limitations.txt]: Done

**Pull request merged**
